pandas/core/generic.py

```diff
@@ -1116,16 +1102,25 @@ def f(x):
         else:
             return result.__finalize__(self)
 
-    def rename_axis(self, mapper, axis=0, copy=True, inplace=False):
+    def rename_axis(self, mapper=None, **kwargs):
```

Comment:
We went to some lengths to preserve the correct function signature in DataFrame.rename. See `rewrite_axis_style_signature` in `pandas/util/_decorators.py`. Can you see if something similar is possible here.

Reply:
I was able to do it.  Had to modify some tests to properly test that we are generating warnings.